content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,367 @@
+# Consensus Algorithms
+
+This document explains, at a high level, how the different consensus algorithms work. After reading it, the engineer should understand the mechanism behind Flare's consensus mechanisms.
+
+## FBA and Separating Sybil Attack Resistance from Consensus by Making it a Local Selection
+
+### Consensus
+
+A consensus algorithm is an agreement system that follows a predefined set of steps in order to allow a group of participants (nodes) to reach to the same decision about a statement.
+In a blockchain network nodes repeatedly decide what the complete history of a shared ledger is, from among multiple possible versions that occasionally conflict between each other.
+This network-wide agreement allows the recipient of a crypto coin to have faith that the coin is both valid (not counterfeit) and not already spent elsewhere.
+
+### Federated Byzantine Agreement
+
+#### Introduction
+
+Any agreement system in a distributed computing network needs to be fault-tolerant.
+This means it must produce consistent results despite errors like slow communication links, unresponsive nodes and misordered messages.
+A Byzantine agreement system is additionally tolerant of [`Byzantine faults`](https://en.wikipedia.org/wiki/Byzantine_fault): nodes that give false information, whether due to error or in a deliberate attempt to subvert the system or gain some advantage.
+Such a system prevents the double spending problem by using a form of majority called _quorum_.
+A node in such a network refuses to commit to a particular version of the ledger history until it sees that enough of its peers are also prepared to commit to it.
+The number of peers that is enough to make a decision is the quorum and once each of the peers commit to a value after its quorum had committed, the system form a large enough voting bloc to force the remaining nodes to agree with the decision.
+In such a system, the attempts from malicious nodes to create double spending events, or in other way invalid transactions, are overwhelmed by the network votes from honest nodes.
+
+For a majority to be formed, the number of all nodes in the system must be known and its quorum calculated at any time.
+Opposed to this concept, in a Federated Byzantine Agreement System (FBAS) the participants in the network are loosely defined and can join and leave at any time without coordinating with any central authority.
```

Comment:
I haven't really seen `FBAS` before, I recommend we stick with FBA and put `in a Federated Byzantine Agreement (FBA) system`.

Reply:
This is one of the ways it is mentioned in the white paper
![image](https://user-images.githubusercontent.com/94193030/145939857-ff0c82af-d3dc-4062-947f-b606c670117c.png)